pulsar-client-cpp/lib/PartitionedConsumerImpl.cc

```diff
@@ -162,14 +174,17 @@ void PartitionedConsumerImpl::handleUnsubscribeAsync(Result result, unsigned int
         callback(ResultUnknownError);
         return;
     }
-    assert(unsubscribedSoFar_ <= numPartitions_);
-    assert(consumerIndex <= numPartitions_);
+    Lock consumersLock(consumersMutex_);
+    const auto numPartitions = numPartitions_;
```

Comment:
Could this change be avoid? Seems brings in lot of un-needed changes, also a little confusing to made `numPartitions_` and `numPartitions` mix-used together in this file.

Reply:
I'll change soon, here I should've called method like `getNumPartitionsWithLock()` (like I did in `PartitionedConsumerImpl`.
Because `producers_`/`consumers_` may be modified in timer's callback, the code that accessed `producers_`/`consumers_` and some other members (see comments in headers) should be protected by lock()/unlock(), except in `start()`.